### ISSUE TITLE
SageAdapter does not pass threads

### ISSUE THREAD

**Issue by jpfeuffer** [OpenMS/OpenMS#7031]

**Comment by timosachsenberg** [comment]
No brainer. Needs to be passed with
```
      --batch-size <batch-size>
          Number of files to search in parallel (default = number of CPUs/2)
```

**Comment by timosachsenberg** [comment]
will add

**Comment by lazear** [comment]
FYI - the `--batch-size` parameter *only* affects how many files are loaded in parallel (e.g. for memory constrained systems). This does not affect how many threads are used by Sage. Which is all of them. If this is necessary, I can add another argument.

**Comment by jpfeuffer** [comment]
That would be great! I think it is always nice to be able to limit the amount of threads a process occupies. Might avoid overcommitting in some situations, right?

**Comment by lazear** [comment]
I don't think it will ever overcommit - the workstealing library I use (Rayon) will default to # of logical CPU cores. I've never had an issue with this, but then again the vast majority of my Sage runs are done in a VM where I want maximum resources to be consumed.

It seems like there is also an environment variable that can be set: `RAYON_NUM_THREADS` (https://docs.rs/rayon/latest/rayon/struct.ThreadPoolBuilder.html#method.num_threads).

Running Sage like `RAYON_NUM_THREADS=2 ./target/release/sage config ...` seems to do the trick
